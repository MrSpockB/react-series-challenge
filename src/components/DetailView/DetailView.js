import React, {Component} from 'react';
import { connect } from 'react-redux';
import {bindActionCreators} from "redux";
import detailGif from "../../actions/detailGif";
import {Container} from "../../App.styled";
import {Gif} from "../Gif/Gif";
class DetailView extends Component {
  constructor(props) {
    super(props);
    const gifId = props.match.params.gifId;
    props.getSingleGifRequested(gifId);
  }
  render() {
    const { gif } = this.props.detailGif;
    return (
      <Container>
        <Gif gif={gif} fullWidth={true}/>
      </Container>
    )
  }
}

const mapStateToProps = (state) => {
  const { detailGif } = state;
  return { detailGif };
};
const mapDispatchToProps = (dispatch) => {
  const { getSingleGifRequested } = detailGif.creators;

  return bindActionCreators({
    getSingleGifRequested,
  }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(DetailView);