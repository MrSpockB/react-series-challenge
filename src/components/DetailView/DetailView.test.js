import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { DetailView } from "./DetailView";

describe('DetailView component', () => {
  it('should render correctly', () => {
    const output = shallow(
      <DetailView
        match={{params: {gifId: 1}, isExact: true, path: "", url: ""}}
        detailGif={{gif: {}}}
        favoritesGifs={{ allGifs: {} }}
        addGif={jest.fn}
        removeGif={jest.fn}
        getSingleGifRequested={jest.fn}
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  })
});