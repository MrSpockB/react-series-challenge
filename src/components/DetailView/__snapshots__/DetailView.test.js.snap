// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailView component should render correctly 1`] = `
<styled.div>
  <Component
    addGif={[Function]}
    fullWidth={true}
    gif={Object {}}
    isFavorited={false}
    isInDetail={true}
    removeGif={[Function]}
  />
</styled.div>
`;
